### ISSUE TITLE
[BUG] New to this program. How does "classes" argument work on COCO dataset?

### ISSUE THREAD

**Issue by DWSuryo** [voxel51/fiftyone#4570]
### Describe the problem

Hello. I'm new at this program so I think I would explore a bit. I want to use COCO 2017 dataset, but with at least has "person" class in each dataset so my goal is to make and export into COCO with person dataset. When I run the code, I thought the classes argument filters as intended, but it resulted with 0 samples. Currently I'm using validation dataset first before doing export for person only COCO dataset

### Code to reproduce issue

Provide a reproducible test case that is the bare minimum necessary to generate
the problem. Please avoid sharing code that relies on your local data or
datasets. Include a short video or screenshot if the bug is in the App.

```
import fiftyone as fo
import fiftyone.zoo as foz

# Load the COCO-2017 validation split with only the "person" class
dataset = foz.load_zoo_dataset(
    "coco-2017",
    dataset_dir='./dataset',
    split="validation",
    label_types=["detections"],
    classes=["person"],
)
# Launch the FiftyOne app to visualize the dataset
session = fo.launch_app(dataset)
```

Output
```
Downloading split 'validation' to './dataset/validation' if necessary
INFO:fiftyone.zoo.datasets:Downloading split 'validation' to './dataset/validation' if necessary
Downloading annotations to './dataset/tmp-download/annotations_trainval2017.zip'
INFO:fiftyone.utils.coco:Downloading annotations to './dataset/tmp-download/annotations_trainval2017.zip'
 100% |██████|    1.9Gb/1.9Gb [2.5s elapsed, 0s remaining, 761.6Mb/s]      
INFO:eta.core.utils: 100% |██████|    1.9Gb/1.9Gb [2.5s elapsed, 0s remaining, 761.6Mb/s]      
Extracting annotations to 'dataset/raw/instances_val2017.json'
INFO:fiftyone.utils.coco:Extracting annotations to 'dataset/raw/instances_val2017.json'
Writing annotations for 0 downloaded samples to './dataset/validation/labels.json'
INFO:fiftyone.utils.coco:Writing annotations for 0 downloaded samples to './dataset/validation/labels.json'
Dataset info written to './dataset/info.json'
INFO:fiftyone.zoo.datasets:Dataset info written to './dataset/info.json'
Loading existing dataset 'coco-2017-validation'. To reload from disk, either delete the existing dataset or provide a custom `dataset_name` to use
INFO:fiftyone.zoo.datasets:Loading existing dataset 'coco-2017-validation'. To reload from disk, either delete the existing dataset or provide a custom `dataset_name` to use
```

Check dataset details:

```
print(dataset)
```

Output:

```
Name:        coco-2017-validation
Media type:  None
Num samples: 0
Persistent:  False
Tags:        []
Sample fields:
    id:           fiftyone.core.fields.ObjectIdField
    filepath:     fiftyone.core.fields.StringField
    tags:         fiftyone.core.fields.ListField(fiftyone.core.fields.StringField)
    metadata:     fiftyone.core.fields.EmbeddedDocumentField(fiftyone.core.metadata.Metadata)
    ground_truth: fiftyone.core.fields.EmbeddedDocumentField(fiftyone.core.labels.Detections)
```

This is using validation only COCO dataset. Since the result is 0 samples, I wonder about how the filtering works in the program. Does the program handle "person" like samples with only "person" class or am I missing something here? I look again at the documentation for [COCO integration](https://docs.voxel51.com/integrations/coco.html), there is "only_matching" argument which interests me. However, using either ```True``` or ```False``` value still results 0 samples. What should I do?

### System information

-   **OS Platform and Distribution** (e.g., Linux Ubuntu 22.04): Google Colab
-   **Python version** (`python --version`): 3.10.12
-   **FiftyOne version** (`fiftyone --version`): 0.24.1
-   **FiftyOne installed from** (pip or source): pip

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another
member of your organization be willing to contribute a fix for this bug to the
FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [ ] Yes. I would be willing to contribute a fix for this bug with guidance
        from the FiftyOne community
-   [x] No. I cannot contribute a bug fix at this time

**Comment by DWSuryo** [comment]
Update: the question still remains about how the class filtering works, currently the workaround I use is to do this:

```
import fiftyone as fo
import fiftyone.zoo as foz

# Load the COCO-2017 train split
train_dataset = foz.load_zoo_dataset("coco-2017", split="train")
# Filter the train dataset to include only samples with the "person" class
train_view = train_dataset.filter_labels("ground_truth", fo.ViewField("label") == "person")

# Load the COCO-2017 validation split
val_dataset = foz.load_zoo_dataset("coco-2017", split="validation")
# Filter the validation dataset to include only samples with the "person" class
val_view = val_dataset.filter_labels("ground_truth", fo.ViewField("label") == "person")

# Launch the FiftyOne app to visualize the filtered datasets
session = fo.launch_app(train_view)

# Define the export directories
train_export_dir = "/path/to/export/train"
val_export_dir = "/path/to/export/validation"

# Export the filtered train dataset in COCO format
train_view.export(
    export_dir=train_export_dir,
    dataset_type=fo.types.COCODetectionDataset,
    label_field="ground_truth"
)

# Export the filtered validation dataset in COCO format
val_view.export(
    export_dir=val_export_dir,
    dataset_type=fo.types.COCODetectionDataset,
    label_field="ground_truth"
)
```

However, with this method, I downloaded all the coco-2017 train and validation dataset, then I filter based on the "person" class label in a "ground_truth". I wonder if there is an efficient way so I can download the images with at least a "person" class in the image

**Comment by joshnguyen08** [comment]
I think I had same issue, here was my workaround, basically download 100 sample then create a view to filter only "person", but I also created a code that looked for "cell phone" and that worked perfectly fine, not sure why there's an issue with "person" only

`import fiftyone as fo
import fiftyone.zoo as foz

# Load the full COCO-2017 dataset without filtering by class
dataset = foz.load_zoo_dataset(
    "coco-2017",
    split="train",
    label_types=["detections"],
    max_samples=100,  # Download 100 samples for testing
    dataset_name="coco-2017-full"
)

# Manually filter the dataset for samples that contain the "person" label
view = dataset.filter_labels("ground_truth", fo.ViewField("label") == "person")

# Print the number of samples with 'person'
print(f"Total samples with 'person': {len(view)}")

# Launch the FiftyOne App to visualize the filtered dataset
session = fo.launch_app(view, port=5151)

# Blocks execution until the app is closed
session.wait()
`

Output:
`Dataset 'coco-2017-full' created
Total samples with 'person': 51
Total samples: 100
App launched. Point your web browser to http://localhost:5151`

**Comment by brimoor** [comment]
Hi all 👋 

`fiftyone==0.24` introduced a bug that broke the `classes` argument for COCO-201X when passed to `load_zoo_dataset()`:

```py
dataset = foz.load_zoo_dataset("coco-2017", ..., classes=["this", "doesnt", "work", 😢], ...)
```

It's patched in https://github.com/voxel51/fiftyone/pull/4884 and will be available as of `fiftyone>=1.0.1` when it is released.

In the meantime, you can omit the `classes` argument from `load_zoo_dataset()` and instead filter after loading the data into FiftyOne via `filter_labels()`!

```py
from fiftyone import ViewField as F

view = dataset.filter_labels("ground_truth", F("label") == "person")
```
